### ISSUE TITLE
Not all prefix parameters specified

### ISSUE THREAD

**Issue by egoholic** [chingor13/json_api_client#166]
I have relation ```belongs_to``` for all parent association but I got the "Not all prefix parameters specified" error when try to create/update entity. Could anyone explain how I should define associations to make POST request on ```entityAs/:entityA_id/entityBs/entityB_id/entityCs/entityC_id/entityDs```?

Now for EntityD I have:

```
  belongs_to :entityA
  belongs_to :entityB
  belongs_to :entityC
```

For now I can successfuly get EntityDs of EntityC, but I can't create and update them.

**Comment by JohnSmall** [comment]
I've got this problem as well. I can't work out how to add records that have a belongs_to relationship defined in the class.

**Comment by Genkilabs** [comment]
Yep, even in the more simple case of _Content_ belongs_to _ContentType_, I can include the :content_type_id in my created _Content_ < JsonApiClient::Resource object, and it will correctly route to ```.../content_types/:content_type_id/contents``` on a POST. However the default behavior seems to not include actually setting the content_type_id attribute on the created object.

Either it creates params like ```{"data"=>{"type"=>"contents", "attributes"=>{"content-type-id"=>1, "status"=>"Live"}}, "content_type_id"=>"1"}``` as seen in https://github.com/chingor13/json_api_client/issues/258 which simply returns "Param not allowed"
**OR**
If you exclude the foreign key in the 'attributes' hash, then the default behavior now ignores the relation and adds an error to the created object's errors hash like ```{"title"=>"must exist", "detail"=>"content-type - must exist", "code"=>"100", "source"=>{"pointer"=>"/data/relationships/content-type"}, "status"=>"422"}```

**Comment by Hitesh-bigscal** [comment]
Same issue for me when I use polymorphic association.

**Comment by adamdullenty** [comment]
Is there any solution or workaround for this please? We just upgraded to 1.5.3 from an early version of the gem and started seeing this issue.

**Comment by senid231** [comment]
we have this issue fixed in `1.6.0`

to make it work in `1.5.3` you need to add prefix parameter to `read_only_attributes`

@Genkilabs in your case it could be something like:

```ruby
class BaseResource < JsonApiClient::Resource
  self.site = "http://example.com/"
end

class ContentType < BaseResource
end

class Content < BaseResource
  belongs_to :content_type
  self.read_only_attributes = read_only_attributes + [:content_type_id]
end
```

**Comment by senid231** [comment]
@egoholic you have a very strange case but if I think if you add '[:entity_a, :entity_b, :entity_c]` to `read_only_attributes` it will work too
